posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,322 @@
+
+* Test applications;
+* The actual tests;
+* Server configuration files used by the tests;
+
+In a similar fashion, application developers can use the Eclipse Transformer to update their applications and test artifacts.
+
+The transformer is quick and efficient.  When integrated as a build step, as is done in Open Liberty builds, using the transformer avoids the creation of new Jakarta EE 9 source files.
+
+Since Jakarta EE 9 is the same as Jakarta EE 8, but with different packages, this article will refer to Jakarta EE 8 applications and Open Liberty features. The article is also applicable to Java EE 7 applications and Open Liberty features.
+
+== Test Transformation Strategy
+
+The stratgy used for Jakarta EE 9 test enablement in Open Liberty is to annotate test classes and methods to indicate which of these are to be run as Jakarta EE 9 tests.  The Open Liberty test framework transforms and re-runs the annotated tests as Jakarta EE 9 tests.
+
+Two types of updates are required when transforming test artifacts.  First, package references must be updated, changing old "javax" package references into new "jakarta" package references.  Package references can be in several locations, including test applications, test code, and server configurations.  Second, server configurations must be updated to replace old Jakarta EE 8 features with corresponding new Jakarta EE 9 features.
+
+Open Liberty updates package references using using the Eclipse Transformer.
+
+The Open Liberty test framework updates the feature names in the server configurations using existing function in the framework. Alternatively, the transformer could be used to update features in the server configuration if rules data were created with the substitution values needed for this update. Application and test developers will need to create code similar to the test framework code, or will need to update server configurations manually.
+
+== Invoking the Transformer
+
+The Eclipse Transformer can be used directly from the command line, as a build task, or from Java code.  This article describes using the transformer from the command line.
+
+(See the link:https://projects.eclipse.org/projects/technology.transformer[transformer project pages] for detailed usage guidelines.)
+
+=== Downloading the Transformer 
+
+Top level link:https://projects.eclipse.org/projects/technology.transformer/downloads[downloads links] are available within the project pages.
+
+Several current link:https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/[snapshots] are usually available.
+
+=== Building the transformer
+
+Alternatively, the transformer can be built by cloning the link:https://github.com/eclipse/transformer[Eclipse Transformer repository] and invoking maven to build the transformer jar:
```

Comment:
Does using maven mean you don't have to do the steps in the "Preparing the work area" section? If so recommend combining that section  with the "Downloading the transformer" and providing the maven info after as an alternative.

Reply:
I revised this and the following sections.  See my update.